src/controllers/auth.js

```diff
@@ -155,129 +146,132 @@ export function logUserIn(attrs) {
     });
 };
 
+export function sendResetToken(user) {
+    const token = uuid.v4();
+
+    return new Promise((resolve, reject) => {
+        PasswordResetToken.where({user_id: user.id})
+            .fetch()
+            .then((model) => {
+                if (!model) {
+                    return new PasswordResetToken({
+                        user_id: user.id,
+                        reset_token: token
+                    }).save();
+                } else {
+                    return model.save({reset_token: token}, {patch: true});
+                }
+            }).then(() => {
+                return sendMail({
+                    to: user.email,
+                    subject: 'Reset your Kitchen Support password',
+                    text: `Hey there, you seem to have requested a password reset. Click on the link below to enter your new password! Warning: the link is only active for 30 minutes from the time that this email is sent!\n\nhttp://kitchen.support/#/forgot-password/${token}\n\nIf this wasnt you, you can disregard this email.`
+                });
+            }).then(resolve).catch(reject);
+    });
+};
+
+export function resetPassword(req) {
+    const {password, reset_token} = req;
+
+    return Promise.all([
+        PasswordResetToken.where('reset_token', reset_token).fetch({
+            withRelated: 'user'
+        }),
+        User.hashPassword(password)
+    ]).then((values) => {
+        if (!values[0]) {
+            throw new Error('Invalid reset token');
+        }
+
+        const expireDate = new Date(values[0].get('expire_date'));
+        const newPassword = values[1];
+
+        return values[0].destroy().then((resetToken) => {
+            if (new Date() > expireDate) {
```

Comment:
Not sure if this is an issue, but the database time could be in a different time zone, so maybe get the current time with the knex command?